src/node_platform.cc

```diff
@@ -269,21 +274,18 @@ void PerIsolatePlatformData::Shutdown() {
   CHECK_NULL(foreground_tasks_.Pop());
   CancelPendingDelayedTasks();
 
+  ShutdownCbList* copy = new ShutdownCbList(std::move(shutdown_callbacks_));
+  flush_tasks_->data = copy;
   uv_close(reinterpret_cast<uv_handle_t*>(flush_tasks_),
            [](uv_handle_t* handle) {
+    ShutdownCbList* callbacks = static_cast<ShutdownCbList*>(handle->data);
+    for (const auto& callback : *callbacks)
+      callback.cb(callback.data);
```

Comment:
Memory leak? Shouldn't you be calling `delete callbacks` here?

Reply:
Yup, thanks for noticing, done.